controller/scheduler/job.go

```diff
@@ -99,6 +99,8 @@ type Job struct {
 
 	// hostError is the error from the host if the job fails to start
 	hostError *string
+
+	serviceFirstSeen *time.Time
```

Comment:
This is set [here](https://github.com/flynn/flynn/blob/scheduler-service-events/controller/scheduler/scheduler.go#L1476-L1479) and used [here](https://github.com/flynn/flynn/blob/scheduler-service-events/controller/scheduler/scheduler.go#L1492-L1496) to enforce a timeout from when the service is first seen to marking the job as running.